formatter/src/main/java/org/togetherjava/formatter/Formatter.java

```diff
@@ -0,0 +1,319 @@
+                      .append("\n```");
+            } else {
+                result.append(revert(section.tokens()));
+            }
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Formats the given string using a given lexer
+     *
+     * @param input input to format
+     * @param lexer lexer to use
+     * @return resulting code
+     * @author illuminator3
+     */
+    public String format(String input, Lexer lexer) {
+        return format(lexer.tokenize(input));
+    }
+
+    /**
+     * Reverts the tokens into the "original state" before the tokenization occurred
+     *
+     * @param tokens tokens to revert
+     * @return original state
+     * @author illuminator3
+     */
+    private String revert(List<Token> tokens) {
+        return tokens.stream().map(Token::content).collect(Collectors.joining());
+    }
+
+    /**
+     * Writes and formats a given code section (in form of a list of tokens) into a StringBuilder
+     *
+     * @param tokens tokens to write
+     * @return written code sections
+     * @author illuminator3
+     */
+    private StringBuilder writeCodeSection(List<Token> tokens) {
+        purgeWhitespaces(tokens = makeMutable(tokens));
+
+        TokenType lastToken = TokenType.UNKNOWN;
+        StringBuilder result = new StringBuilder();
+        boolean lastNewLine = false;
+        int indentation = 0;
+        boolean forClosed = true;
+        int forClosingLevel = 0;
+
+        for (Token token : tokens) {
+            TokenType type = token.type();
+
+            if (lastNewLine && type != TokenType.CLOSE_BRACES) {
+                putIndentation(indentation, result);
+            }
+
+            lastNewLine = false;
+
+            iteration: {
+                if (type == TokenType.OPEN_BRACES) {
+                    if (lastToken == TokenType.CLOSE_PARENTHESIS
+                            || lastToken == TokenType.IDENTIFIER || lastToken == TokenType.TRY || lastToken == TokenType.ELSE) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+                    indentation++;
+
+                    break iteration;
+                } else if (lastToken == TokenType.CLOSE_BRACKETS && type != TokenType.SEMICOLON) {
+                    result.append(" ");
+
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.CLOSE_BRACES) {
+                    result.append("\n");
+
+                    putIndentation(--indentation, result);
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+
+                    break iteration;
+                } else if (type == TokenType.IDENTIFIER && (lastToken == TokenType.CLOSE_BRACES
```

Comment:
Those conditions are really hard to read, can you split them up into a function?
```java
else if(whateverThatIfMeans(lastToken, token)
```